### PR TITLE
ZScaler: Feature - Added 4 new actions

### PR DESCRIPTION
**Pull request by kelby-shelton**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
Added 4 new actions
#### get users
Gets a list of users based on name, department or group
#### get groups
Gets a list of groups based on name
#### add user to group
Adds a user to a group based on user id and group id
#### remove user from group
Removes a user to a group based on user id and group id

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
4 new actions

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by jdemelo** [comment]
Looks pretty good, I think we just need to fix the static test failures, and for some reason the integration test results are missing from this PR.

**Review by pdesai-crest — APPROVED** [review]

**Review by mishalp-crest — APPROVED** [review]

File: zscaler_connector.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def _get_error_message_from_exception(self, e):
                 elif len(e.args) == 1:
                     error_msg = e.args[0]
         except Exception:
-            pass
+            self.debug_print("Error occurred while getting message from response")
```
Comment (paulnawat) [inline zscaler_connector.py:67]: Should we include the error message in the print? This should get us to the error faster.

File: zscaler_connector.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def _process_html_response(self, response, action_result):
             split_lines = error_text.split('\n')
             split_lines = [x.strip() for x in split_lines if x.strip()]
             error_text = '\n'.join(split_lines)
-        except:
+        except Exception:
             error_text = "Cannot parse error details"
```
Comment (paulnawat) [inline zscaler_connector.py:122]: Can we include the error message in the text or in debug_print?

File: zscaler_connector.py [hunk 3]
```diff
@@ -822,6 +823,169 @@ def _handle_get_admin_users(self, param):
...
+            "dept": param.get('dept'),
+            "group": param.get('group'),
+            'page': 1
+        }
+        users = []
+        while True:
+            if limit < ZSCALER_MAX_PAGESIZE:
+                params['pageSize'] = limit
+            else:
+                params['pageSize'] = ZSCALER_MAX_PAGESIZE
```
Comment (paulnawat) [inline zscaler_connector.py:857]: suggestion:
```python
params['pageSize'] = min(limit, ZSCALER_MAX_PAGESIZE)
```

File: zscaler_connector.py [hunk 4]
```diff
@@ -822,6 +823,169 @@ def _handle_get_admin_users(self, param):
...
+            ret_val, get_users = self._make_rest_call_helper('/api/v1/users', action_result, params=params)
+            if phantom.is_fail(ret_val):
+                return action_result.get_status()
+            for user in get_users:
+                users.append(user)
```
Comment (paulnawat) [inline zscaler_connector.py:862]: suggestion:
What do you think about this variable names?
get_users => users or current_users
users => all_users

**Review by paulnawat — APPROVED** [review]
lgtm. I left some suggestions.

File: zscaler_connector.py (hunk 3 above)
Comment (mmomin-crest) [reply zscaler_connector.py:857]: Updated code as per suggestion.

File: zscaler_connector.py (hunk 2 above)
Comment (mmomin-crest) [reply zscaler_connector.py]: Yes, Added into debug_print.

**Pull request merged**